New Usage Guide and V1 of a checklist (#100)
scarf-sh/docs: docs/user_best_practices.md

> # Scarf User Guide & Best Practices

Reviewer: ```suggestion
# User Guide & Best Practices

```

## Intro

This is a user guide to best practices when using Scarf.  Consider this not only a guide but also an FAQ.  

Currently this guide contains:

> 1. How to use Scarf Data in existing  sales and marketing efforts

Reviewer 2: It would be amazing if you could link this table of contents to the sections below

Author: Will do in an update.

	1. Events you should be tracking
	2. The goal is ongoing usage 
	3. How to use Scarf to infer ongoing usage 
	4. Basic lead scoring with Scarf
	5. Sample customer journey 
	6. What to do with companies identified by Scarf
	7. Common ways to inject Scarf into existing sales/marketing activities 
	8. Common ways for startups to build their first sales and marketing processes 
2. How to use confidence intervals
3. Best Practices to track multiple projects and packages
	1. Handling different projects
	2. Handling different versions
	3. Handling different platforms
	4. Container best practices
	5. Variables 
    6. Pixels for each project	
4. Tracking external link clicks

## How to use Scarf data in existing sales and marketing efforts

### Events we should be tracking:

> There are 3 events I would suggest everyone track.  

Reviewer 2: Instead of "I", should likely be "we" since we're speaking from the perspective of the Scarf organization here.

```suggestion
There are 3 events we would suggest everyone track. 
```

> - Downloads/Pull events

Reviewer 2: Might be nice to add section headers then link to them here.

### The Riskiest But Most Valuable Metric: Ongoing Usage

While the three activities above are straightforward and generally not viewed with too much concern, there is a fourth activity or metric you can (and probably should) track.  An essential, albeit controversial, activity that serves as a highly valuable metric for any organization seeking to understand the usage patterns of its software - the use of 'call-home' functionality, also known as ongoing usage tracking. The call-home functionality is a mechanism within your software that sends a signal, or a 'ping', back to a designated server or gateway. This signal provides you with real-time information about your software's usage in live production environments, surpassing the insight level gained from just tracking downloads.

While download data can indicate interest and repeated use of your software, the ongoing, consistent 'ping' or call-home activity serves as a definitive predictor of your software's actual usage. Consider this the 'Nirvana' of metrics for your projects, the golden standard that allows you to measure the exact magnitude of your active install base and the frequency of software usage and deployment.

> However, implementing this mechanism requires a degree of technical adaptation. Platforms like Scarf, for instance, offer this capability out of the box. But to make full use of it, you'll need to adjust your application accordingly. There are different ways to accomplish this; for JavaScript applications, a package called 'Scarf-JS' can be used. Alternatively, a lightweight, background 'ping' or activity back to a Scarf gateway event can be employed. This ping can be triggered when your application starts up, is used, or at any other specified event.

Reviewer 2: ```suggestion
However, implementing this mechanism requires a degree of technical adaptation. Platforms like Scarf, for instance, offer this capability out of the box. But to make full use of it, you'll need to adjust your application accordingly. There are different ways to accomplish this; for JavaScript applications, a package called [`scarf-js`](https://www.npmjs.com/package/@scarf/scarf) can be used. Alternatively, a lightweight, background 'ping' or activity back to a Scarf gateway event can be employed. This ping can be triggered when your application starts up, is used, or at any other specified event.
```

### Customer journey example with Scarf
| Different Phases of Interest                                               | Description                                                                                                                                        | Events                                                                                                                                                                 | Action                                                                                                                                                                              |
|----------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | Passive interest: <br>Hello World                                               | Someone discovered or visited your website. They may or may not have any interest in your software or projects.                                      | Web traffic to docs or websites over the course of 1 or 2 days.                                                                                                         | I would not take any action here.                                                                                                                                                   |

Reviewer 2: Might be good to highlight titles/keys as bolded in the next couple sections

```suggestion
| **Passive interest:** <br>Hello World                                               | Someone discovered or visited your website. They may or may not have any interest in your software or projects.                                      | Web traffic to docs or websites over the course of 1 or 2 days.                                                                                                         | I would not take any action here.                                                                                                                                                   |
```

| Waning Interest &amp; Potential Churn: <br>Uh oh… this user is at risk          | Usage is dropping and there is risk that this user may turn from an active user to a former user.                                                  | If you see massive drop offs in traffic and downloads over a 30 day period this sends up red flags.                                                                    |

### What to do with companies identified by Scarf

Effectively employing this data requires a strategic and tailored approach to meet the unique needs and goals of each organization. Below are some general recommendations that apply to most situations:

> - Understand Your Audience: Use download data and website traffic information to build a deeper understanding of your audience. This involves analyzing who is downloading your software, viewing your documentation, and browsing your website. With this information, you can enrich your existing leads, score potential ones, and build a well-informed customer profile.

Reviewer 2: could also bold the titles/keys in this section

For startups or companies initiating new sales and marketing initiatives, creating a lightweight growth engine that nurtures potential users can be the key to driving growth. Setting up a lifecycle or nurture campaign can guide potential users through your marketing funnel, providing them with the right content at the right time to foster interest and engagement.

Promoted content can be a powerful tool in these campaigns. Aimed at users in the early stages of engagement, this content can educate users about your software, showcasing its features and benefits and encouraging them to explore it further. As these potential users turn into ongoing users, you can begin to introduce promoted content, offers, and cold outreach to convert them into paying customers.

Understanding the customer journey is crucial in a startup or new marketing environment. By mapping out this journey and identifying combinations of events and thresholds, you can strategize when to increase or decrease marketing activities for optimal effect. This dynamic approach can keep your marketing efforts agile and responsive to user behavior.

> Social media platforms like LinkedIn offer a targeted way to reach specific companies.

Reviewer 2: ```suggestion
Social media platforms like [LinkedIn](https://www.linkedin.com/company/scarf-sh/) offer a targeted way to reach specific companies.
```

- Use this data to build a lightweight marketing and growth engine.
- Approach the process as a life cycle or nurture-type campaign. Nurture potential users until they become productive users.
- Use promoted content targeted towards companies that are downloading or have looked at your documentation.
- Once users are actively using your software, shift the focus to ongoing maintenance and new releases. Then, start introducing your paid offerings or services.
- Use social media to engage potential users, specifically for companies you identified
- Integrate the scarf platform into your existing community activity to help nurture and guide potential users.

> ## How to use confidence intervals

Reviewer 2: ```suggestion

## How to use confidence intervals
```

In your Scarf dashboards you will often see a confidence flag associated with events and companies.

![Confidence.png](./assets/pics/user-guide/Medium_Confidence_Example.png)

The confidence is a measurement of Scarf's confidence in the IP/metadata -> organization match for each event. Some of our metadata providers like Clearbit provide their own confidence scores and Scarf will take those into account, but we also account for what other providers say. We will make our own adjustments in many cases, for instance, if there is disagreement between the different data providers we use, or if we find irregularities in the metadata.

> Confidence intervals have 2 data points associated with them. The first is the overall flag which is low ( red ), medium ( yellow), and green ( high confidence). This gives you a quick way to associate high-probability matches with low ones. When looking into the flags, you will find a % of confidence associated with each company and event, and these percentages can be used to further differentiate which companies and events to prioritize.

Reviewer 2: ```suggestion
Confidence intervals have 2 data points associated with them. The first is the overall flag which is low (red), medium (yellow), and high (green) confidence. This gives you a quick way to associate high-probability matches with low ones. When looking into the flags, you will find a percent (%) of confidence associated with each company and event, and these percentages can be used to differentiate further which companies and events to prioritize.
```

| High     |  The events and organizations that have a high confidence interval mean that we have multiple high probability indicators of a match to an organization for these events. |  High confidence matches provide the highest quality data. We are confident in people using this data for their sales pipelines, lead scoring, etc. These records could be integrated into exsiting tools to give a complete picture of a customers journey |

> <a id="MultiplePackages"></a>

Reviewer 2: what is this for?

Author: This is code for a linked table of contents, will release in the next iteration

> ## Best practices to track multiple packages or projects

Reviewer 2: ```suggestion

## Best practices to track multiple packages or projects
```

Many of our customers offer downloads of multiple different open source projects, different versions, or downloads for different platforms. There are some best practices we have found useful for many of our customers when faced with dozens or even hundreds of unique packages. Your decisions on how to handle multiple packages, projects, or versions ultimately will come down to a.) how you want to report the data and b.) personal preference. 

> Keep in mind you can export data and report on a variety of metrics by package, variable, etc. However, if you want to get quick analytics using the Scarf dashboards, reporting is done for all your packages ( global ), an individual package, or a specific pixel. You can see metrics for each variable under a package or pixel. As of July 2023, variable reporting on a package or pixel is limited to a single dimension. i.e.  I can see a report for {variable1} or {variable2} for a package or pixel, but I can not see data for {variable1} correlated with {variable2}. You can, of course do this correlation and analysis by exporting and using your own analytics tool. Keep this in mind as you build your packages and routes.  

Reviewer 2: ```suggestion
Keep in mind you can export data and report on a variety of metrics by package, variable, etc. However, if you want to get quick analytics using the Scarf dashboards, reporting is done for all your packages (global), an individual package, or a specific pixel. You can see metrics for each variable under a package or pixel. As of July 2023, variable reporting on a package or pixel is limited to a single dimension. i.e.  I can see a report for `{variable1}` or `{variable2}` for a package or pixel, but I can not see data for `{variable1}` correlated with `{variable2}`. You can, of course, do this correlation and analysis by exporting and using your own analytics tool. Keep this in mind as you build your packages and routes.  
```

Some people have found it easier to create routes for new major versions, but you could also use multiple variables for this. I.e.

* `http://companyname.gateway.scarf.sh/projectname/V1/file.{minorversion}.gz`
* `http://companyname.gateway.scarf.sh/projectname/V2/file.{minorversion}.gz`
* `http://companyname.gateway.scarf.sh/projectname/{majorversion}/file.{minorversion}.gz`

Depending on your needs, you could create a new package for each major version, but it is not required.

### Handling different platforms 

Similar to handling different versions, you can use variables if you are shipping different package types or for different platforms. So instead of:

`http://companyname.gateway.scarf.sh/projectname/RHEL5/{majorversion}/file.{minorversion}.gz` and 
`http://companyname.gateway.scarf.sh/projectname/RHEL6/{majorversion}/file.{minorversion}.gz`

You could do the following:

`http://companyname.gateway.scarf.sh/projectname/{osversion}/{majorversion}/file.{minorversion}.gz`

### Containers 

> For containers, we recommend setting up a "Collection" to sit in front of your entire namespace, i.e., `company/*` on Docker Hub or your preferred container registry. Collections automatically sync and keep up to date with container registries making it easy to release new versions without having to worry about Scarf being up to date with new releases. To learn more, check out the Docs for collections here: https://docs.scarf.sh/gateway/#creating-collections

Reviewer 2: ```suggestion
For containers, we recommend setting up a "Collection" to sit in front of your entire namespace, i.e., `company/*` on Docker Hub or your preferred container registry. Collections automatically sync and keep up to date with container registries making it easy to release new versions without having to worry about Scarf being up to date with new releases. To learn more, check out the [collections docs](/gateway/#creating-collections)
```

Create 4 pixels:

- General traffic
	- Embed the general traffic pixel on all pages.  
- high value 
	- Embed the high value on your pricing pages, support pages, and other highly desirable docs/website pages.  
- medium value
	- Embed the medium value pixel on your installation and setup docs, tutorials, or other mid-level docs. 
- low value
	- Embed the low-value pixel on everything else.  

Now you will be able to classify and score users' activities based on what sort of content they viewed. You can even bake this into your existing lead-scoring system.  

#### Overlapping pixels for tags

You can embed multiple pixels on the same page for different reasons to facilitate more detailed reporting. For instance, you may want to create one pixel for each project, but you may also want to create a pixel for just specific types of content on your website or elsewhere on the web. For instance, maybe you want everyone looking at blogs or tutorials on a critical feature to be tracked and have metrics reported on.  

> 

Reviewer 2: ```suggestion
```
